src/components/ValidatedInput.tsx

```diff
@@ -0,0 +1,39 @@
+import { PropsWithChildren, ReactNode, Ref, forwardRef, useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { InputProps, Input } from 'honorable'
+import FormField from './FormField'
+
+export type ValidationResponse = {error: boolean, message: string} | null
+
+export type ValidatedInputProps = InputProps & PropsWithChildren<{
+  label?: ReactNode
+  hint?: ReactNode
+  validation?: (val: string) => ValidationResponse
+}>
+
+const propTypes = {
+  label: PropTypes.node,
+  hint: PropTypes.node,
+  validation: PropTypes.func,
+}
+
+function ValidatedInputRef({label, hint, validation, onChange, width, ...input} : ValidatedInputProps, ref: Ref<any>) {
+    const [error, setError] = useState(null)
+    const wrappedOnChange = useCallback((e: any) => {
+        if (e.target?.value && validation) setError(validation(e.target.value))
+        if (onChange) onChange(e)
+    }, [onChange, validation])
+    
+    return (
+        <FormField label={label} hint={hint} caption={error?.message} width={width}>
```

Comment:
I really think the error message should be with the hint not the caption.

Reply:
Also the files are not linted sowwy :)

Reply:
That's probably true, will the hint text be highlighted with an error color though?

Reply:
Hint can accept a ReactNode, so it's up to you to color it

Reply:
Although would this look weird with interfaces with multiple inputs all being validated? Caption might be better there